### PR TITLE
Fix Windows Codex login by splitting oversized keyring credentials

### PR DESCRIPTION
**Pull request by mpfaffenberger**
## Summary

Fix `/login openai-codex` failing on Windows with `(1783, 'CredWrite', 'The stub received bad data')` when the serialized OAuth token bundle exceeds a credential entry's size limit.

Windows documents a 2560-byte maximum for `CredentialBlob`; Python keyring's Windows backend writes UTF-16. CLAI previously passed the complete token JSON to a single entry.

- Keep small bundles and existing single-entry logins compatible.
- Split oversized bundles into bounded keyring entries, with a versioned manifest at the existing login entry. Use separate services to avoid Windows keyring's multi-account collision handling.
- Verify chunks before publishing the manifest. Preserve the old login on chunk-write failure, clean up failed chunks, and remove superseded chunks after successful publication.
- Keep all token material in the configured keyring backend, with no plaintext fallback or new dependencies.
- Document storage behavior and add regression tests for the Windows byte limit, Unicode, large-token save/refresh, legacy entries, corrupt/missing chunks, and failed writes.

## Verification

- CLAI tests: **153 passed**.
- CLAI Ruff format/check and strict Pyright (`src tests`): passed.
- Repository-wide Ruff format/check using the repository environment: passed.
- Tests use a fake keyring that rejects blobs larger than 2560 UTF-16 bytes. Local host is macOS; native Windows execution has not been performed.

## Sources

- [Microsoft `CREDENTIALW`: `CredentialBlobSize`](https://learn.microsoft.com/en-us/windows/win32/api/wincred/ns-wincred-credentialw)
- [Python keyring Windows backend](https://github.com/jaraco/keyring/blob/main/keyring/backends/Windows.py)

## Failure boundaries

If a backend reports an error after committing the manifest, the new chunks are retained rather than deleting potentially active credentials. Such uncertain writes can leave unused chunks in the OS keyring. This change does not introduce cross-process locking for simultaneous logins or refreshes.

### CODE REVIEW

File: pydantic-clai2/src/pydantic_clai2/credential_store.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# Windows allows 2560 bytes per credential. Keyring writes UTF-16, so a
+# 600-character chunk fits even if every character uses a surrogate pair.
+_CHUNK_SIZE = 600
+_MAX_BYTES = 2560
+
+
+def _chunk_services(*, value: str) -> list[str]:
+    if not value.startswith(_PREFIX):
+        return []
+    match = re.fullmatch(r'clai-chunks-v1:([0-9a-f]{32}):([1-9][0-9]{0,3})', value)
```
[the remark this answers is not in the document]
Comment (mpfaffenberger) [reply pydantic-clai2/src/pydantic_clai2/credential_store.py:22]: Keeping this bound intentionally. This is Codex OAuth credential storage, not an arbitrary-size blob API. The parser consumes persisted data and eagerly builds a service list; accepting an unbounded count would let a corrupt manifest allocate an arbitrarily large list and trigger unbounded keyring reads. The existing bound permits nearly six million characters of token JSON, far beyond the OAuth bundles involved in this failure. Both writing and reading use the same parser, so an oversized bundle is rejected before any chunks are written. Removing the bound is not needed to fix the Windows 2560-byte per-entry limit.

**Comment by mpfaffenberger** [comment]
Acknowledged that changes to OAuth credential storage require human review; this PR is left open for that review. The chunk-count finding has been answered and resolved with the reason for the defensive bound: https://github.com/pydantic/pydantic-ai-harness/pull/886#discussion_r4031529847. CLAI CI passes on Python 3.11 and 3.14, including the 100% branch-coverage gate. Native Windows execution remains unverified; the regression backend enforces the documented Windows UTF-16 byte limit.
